kedro-datasets/kedro_datasets_experimental/pandas/delta_sharing_dataset.py

```diff
@@ -0,0 +1,164 @@
+        "use_delta_format": False  # Default to not using Delta format
+    }
+
+    def __init__( # noqa: PLR0913
+        self,
+        *,
+        share: str,
+        schema: str,
+        table: str,
+        credentials: dict[str, Any],
+        load_args: dict[str, Any] | None = None
+    ) -> None:
+        """Creates a new instance of ``DeltaSharingDataset``.
+
+        Args:
+            share (str): Share name in Delta Sharing.
+            schema (str): Schema name in Delta Sharing.
+            table (str): Table name in Delta Sharing.
+            credentials (dict[str, Any]): Credentials for accessing the Delta Sharing profile. Must include:
+                - `profile_file` (str): Path to the Delta Sharing profile file.
+            load_args (dict[str, Any], optional): Additional options for loading data.
+                - `version` (int, optional): A non-negative integer specifying the version of the table snapshot to load.
+                  Defaults to None, which loads the latest version. This parameter allows you to access historical versions of the shared table.
+                - `limit` (int, optional): A non-negative integer specifying the maximum number of rows to load.
+                - `use_delta_format` (bool, optional): Whether to use the Delta format for loading data. Defaults to False.
+
+        Raises:
+            DatasetError: If the profile file is not specified in credentials.
+        """
+        self.share = share
+        self.schema = schema
+        self.table = table
+        self.profile_file = credentials.get("profile_file")
+
+        # Merge the provided load_args with DEFAULT_LOAD_ARGS
+        self._load_args = {**self.DEFAULT_LOAD_ARGS, **(load_args or {})}
+        self.version = self._load_args["version"]
+        self.limit = self._load_args["limit"]
+        self.use_delta_format = self._load_args["use_delta_format"]
+
+        if not self.profile_file:
+            raise DatasetError("Profile file must be specified in credentials.")
+
+        self.table_path = f"{self.share}.{self.schema}.{self.table}"
+        self.table_url = f"{self.profile_file}#{self.table_path}"
+
+    def _load(self) -> pd.DataFrame:
+        """Load data from the Delta Sharing shared table into a Pandas DataFrame.
+
+        Returns:
+            pd.DataFrame: Loaded data.
+        """
+        return delta_sharing.load_as_pandas(
+            self.table_url,
+            version=self.version,
+            limit=self.limit,
+            use_delta_format=self.use_delta_format
+        )
+
+    def _save(self, data: pd.DataFrame) -> None:
+        """Saving to Delta Sharing shared tables is not supported.
+
+        Args:
+            data (pd.DataFrame): Data to save.
+
+        Raises:
+            NotImplementedError: Saving to Delta Sharing shared tables is not supported.
+        """
+        raise NotImplementedError("Saving to Delta Sharing shared tables is not supported.")
```

Comment:
This should be `DatasetError` which can be imported from `kedro.io.core` as I see some other datasets where one of the operations are not supported do the same.

Reply:
Additionally, added the requested information to the release notes and the API `.rst` file as you suggested. If there's anything else you need, feel free to let me know!